plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/JdkLdapClient.java

```diff
@@ -106,11 +108,30 @@ public boolean isGroupMember(String searchBase, String groupSearch, String conte
         }
     }
 
+    @Override
+    public Set<String> lookupUserGroups(String searchBase, String searchFilter, String contextUserDistinguishedName, String contextPassword)
+            throws NamingException
+    {
+        try (CloseableContext context = createUserDirContext(contextUserDistinguishedName, contextPassword);
+                CloseableSearchResults search = searchContext(searchBase, searchFilter, context)) {
+            ImmutableSet.Builder<String> groupNames = ImmutableSet.builder();
+            if (search.hasMore()) {
+                Attributes attributes = search.next().getAttributes();
+                Attribute memberOfAttribute = attributes.get("memberof");
+                if (memberOfAttribute == null) {
+                    throw new NamingException("No memberOf attribute found... The ldap group provider requires the memberOf overlay to be enabled.");
+                }
+            }
+            return groupNames.build();
+        }
+    }
+
     private static CloseableSearchResults searchContext(String searchBase, String searchFilter, CloseableContext context)
             throws NamingException
     {
         SearchControls searchControls = new SearchControls();
         searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+        searchControls.setReturningAttributes(new String[] {"memberOf"});
```

Comment:
Will it affect the `lookupUserDistinguishedNames` ?

Reply:
From the JavaDocs it looks like 
```
Specifies the attributes that will be returned as part of the search.
null indicates that all attributes will be returned. An empty array indicates no attributes are returned.
Params:
attrs – An array of attribute ids identifying the attributes that will be returned. Can be null.
See Also:
getReturningAttributes
```

Is there is any problem on returning all the attributes ?